aiida_vasp/parsers/file_parsers/stream.py

```diff
@@ -0,0 +1,118 @@
+"""
+Standard stream parser for the VASP output.
+
+--------------
+A parser that analyses standard output for VASP related notification, warnings and
+errors. Typically this is contained in the scheduler standard output.
+"""
+import re
+
+from parsevasp.stream import Stream
+from aiida_vasp.parsers.file_parsers.parser import BaseFileParser, SingleFile
+
+DEFAULT_OPTIONS = {'quantities_to_parse': ['notifications']}
+
+
+class StreamParser(BaseFileParser):
+    """Parser used for parsing errors and warnings from VASP."""
+
+    PARSABLE_ITEMS = {
+        'notifications': {
+            'inputs': [],
+            'name': 'notifications',
+            'prerequisites': [],
+        }
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(StreamParser, self).__init__(*args, **kwargs)
+        self._stream = None
+        self.init_with_kwargs(**kwargs)
+
+    def _init_with_file_path(self, path):
```

Comment:
Just as a note I think aiida-core is trying abstract the repository and make it not dependent on a file system so at somepoint in the future not all files will have the "path". This can be a general problem, for all file parsers, I guess.

Maybe we should move to working with file objects not paths at some point.

Reply:
Yes, this we should have on our roadmap. I thought about addressing this before, but postponed it. I think it is not super critical, but we should create an issue: https://github.com/aiida-vasp/aiida-vasp/issues/409